src/Http/Routing/test/UnitTests/Matching/DfaMatcherTest.cs

```diff
@@ -294,8 +294,8 @@ public async Task MatchAsync_MultipleCandidates_Logging()
             var endpointDataSource = new DefaultEndpointDataSource(new List<Endpoint>
             {
                 CreateEndpoint("/One", 0),
-                CreateEndpoint("/{p:int}", 0),
-                CreateEndpoint("/x-{id}-y", 0),
+                CreateEndpoint("/{p:int}", 1),
+                CreateEndpoint("/x-{id}-y", 2),
```

Comment:
I don't get it. What's the bug?

Reply:
See the linked issue. It is related to the order that the matcher evaluates routes. I'm not 100% certain of the cause because I can't reproduce it locally.

Reply:
but don't these have different precedence? I've written whole novels about how this works. https://github.com/aspnet/AspNetCore/blob/master/src/Http/Routing/src/Matching/EndpointComparer.cs

Reply:
I'm stumped. The test is clearly failing on the CI server, but passes everywhere else.

Reply:
I'm stumped. The test consistently fails on the CI server, but passes everywhere else.

Reply:
How about removing the constraint, that will give them all different precedence - which is what I intended when I wrote the test.

Reply:
I've got an idea:

`ComparePattern` is using `CompareTo`, which is culture sensitive. Perhaps there is something odd about the culture the tests are running under on the CI server.

Could change it to `string.Compare(routeEndpointX.RoutePattern.RawText, routeEndpointY.RoutePattern.RawText, StringComparison.OrdinalIgnoreCase)` and see whether that fixes the order on the CI server.

https://github.com/aspnet/AspNetCore/blob/master/src/Http/Routing/src/Matching/EndpointComparer.cs#L54-L75

Reply:
AH yeah. You're right, I think it could be a legit bug then. I owe you a dollar next time you come to redmond.

Reply:
I'll create a PR with the invariant sort in master.

This current test workaround fix can go into 2.2 🙏